source/PlayerInfo.cpp

```diff
@@ -2774,6 +2799,35 @@ void PlayerInfo::AddStock(const Outfit *outfit, int count)
 
 
 
+// Transfer ships from the player to the planet or vice versa.
+// If needed, create a no-outfit version of this ship if we don't already have one.
+void PlayerInfo::AddStock(const Ship *model, const int count)
+{
+	shipStock[model] += count;
+
+	if(shipStock[model] <= 0 || stockShips.find(model) != stockShips.end())
+		return;
+
+	stockShips[model] = make_shared<Ship>(*model);
+	Ship &ship = *stockShips[model];
+
+	// Make a copy so we can remove all the outfits safely.
+	map<const Outfit *, int> outfits = ship.Outfits();
+	for(const auto &it : outfits)
+		ship.AddOutfit(it.first, -it.second);
```

Comment:
The safety is so we can loop over everything safely while we remove entries.  I doubt it's safe to use a ranged for over a map when you're deleting entries, since it invalidates the pointer to the current element.  I haven't tried it, admittedly.

Reply:
I was just thinking of a ship GetHull() method but yeah we dont need it rn

as for the for I'm not sure either but this seems like a good safety to me